cli/azd/pkg/azapi/azure_resource_types.go

```diff
@@ -130,6 +131,8 @@ func GetResourceTypeDisplayName(resourceType AzureResourceType) string {
 		return "Machine Learning Endpoint"
 	case AzureResourceTypeMachineLearningConnection:
 		return "Machine Learning Connection"
+	case AzureResourceTypeRoleAssignment:
+		return "Role Assignment"
```

Comment:
Just curious, what pushes us towards surfacing role assignments in the provisioning progress here?

Observing the progress after the change, I can see the role assignments do show up here, but the presentation of GUIDs seem a little unsatisfying. 
![image](https://github.com/user-attachments/assets/bf410798-765d-4084-b79b-672d2db91958)

Reply:
Nothing really, I'm fine with leaving it out. Open to others' thoughts too!